pkg/agent/apiserver/handlers/networkpolicy/handler.go

```diff
@@ -16,52 +16,72 @@ package networkpolicy
 
 import (
 	"encoding/json"
+	"fmt"
 	"net/http"
+	"net/url"
+	"strings"
 
-	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
+	agentquerier "github.com/vmware-tanzu/antrea/pkg/agent/querier"
 	cpv1beta1 "github.com/vmware-tanzu/antrea/pkg/apis/controlplane/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/querier"
 )
 
 // HandleFunc creates a http.HandlerFunc which uses an AgentNetworkPolicyInfoQuerier
 // to query network policy rules in current agent.
-func HandleFunc(aq querier.AgentQuerier) http.HandlerFunc {
+func HandleFunc(aq agentquerier.AgentQuerier) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		name := r.URL.Query().Get("name")
-		ns := r.URL.Query().Get("namespace")
-		pod := r.URL.Query().Get("pod")
-
-		if (name != "" || pod != "") && ns == "" {
-			http.Error(w, "namespace must be provided", http.StatusBadRequest)
+		npFilter, err := NewFilterFromURLQuery(r.URL.Query())
```

Comment:
If not reused out of the package (and it's likely this case since this is specific to handler), could we make `NewFilterFromURLQuery` private?

Reply:
Yeah, thx for your suggestion.